### PR TITLE
Add storybook story for Disabled component

### PR DESCRIPTION
**Pull request by brentswisher**
## Description
Adds a storybook story for the Disabled component as part of #17973

I had to make a decision on this story regarding what/how much to put in the component to show as disabled. I included all the "basic" form controls as well as a few of the more complex ones. I don't have a strong feeling on what should be shown, so if anyone has feedback on more/less child components just let me know.

## Concerns
I also ran into a few things with the storyshot integration I want to call out:
* It seems including another component in a new story increments an identifier used in the other stories:
Comparing_master___add_disabled-component-story_·_WordPress_gutenberg_1 
I am pretty sure this is intentional but didn't see it documented everywhere so I just wanted to double-check.
* The Disabled component uses a MutationObserver which jest doesn't support. Using the unit test for the Disabled component as a template I added a stub of window.MutationObserver in beforeAll() and removed it in afterAll(). This appears to work but doesn't seem ideal as none of the other storybook tests need it. If there is a better way to do this feel free to point me in that direction.

## How has this been tested?
* run `npm run storybook:dev`
* See the new Disabled story under components

## Screenshots
Components___Disabled_-_Default_⋅_Storybook

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by gziolo** [comment]
As discussed on Slack yesterday, we will need to upgrade Jest to use the latest version of JSDom which supports `MutationObserver`. I guess you can add a code comment and create a follow-up issue if we don't want to wait with this PR.

**Comment by brentswisher** [comment]
I updated jest, jsdom and storybook to the most recent versions locally and am still getting this error without the setup/teardown methods:

`● Storyshots › Components/Disabled › Default

    TypeError: window.MutationObserver is not a constructor

      47 | 		this.disable();
      48 |
    > 49 | 		this.observer = new window.MutationObserver( this.debouncedDisable );
         | 		                ^
      50 | 		this.observer.observe( this.node, {
      51 | 			childList: true,
      52 | 			attributes: true,`

The js configuration for Gutenberg is a bit more complex than I am used to working with, so it's possible I am not updating what I think I am or in the correct place.

**Comment by gziolo** [comment]
I opened #20766 which tries to upgrade Jest and JSDom to the more recent versions.

File: storybook/test/index.js [hunk 1]
```diff
@@ -6,6 +6,20 @@ import initStoryshots, {
 } from '@storybook/addon-storyshots';
 import path from 'path';
 
+let MutationObserver;
```
Comment (gziolo) [inline storybook/test/index.js]: Can you add a comment referencing the PR I opened to update `jest` and `jsdom`?

File: packages/components/src/disabled/stories/index.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import AnglePickerControl from '../../angle-picker-control/';
+
+export default {
+	title: 'Components/Disabled',
+	component: Disabled,
+};
+
+export const _default = () => {
+	return (
+		<Disabled>
```
Comment (gziolo) [inline packages/components/src/disabled/stories/index.js]: Do we need all those components to include in here? Can we use the most representative 3? I don't have answers, I'm thinking whether it's enough to cover those that use `input`, `select` and `textarea`.

**Review by gziolo — COMMENTED** [review]
It all looks great. I left two comments to respond. We should document the issue with `MutationObserver` and file a follow-up issue to unblock this PR.

File: packages/components/src/disabled/stories/index.js (hunk 2 above)
Comment (brentswisher) [reply packages/components/src/disabled/stories/index.js]: I questioned that too, and nearly added just a single `input`. I would be fine with making it just the top three, maybe @ItsJonQ might have an opinion?

File: storybook/test/index.js (hunk 1 above)
Comment (brentswisher) [reply storybook/test/index.js]: Added, thanks for the review

File: packages/components/src/disabled/stories/index.js (hunk 2 above)
Comment (ItsJonQ) [reply packages/components/src/disabled/stories/index.js]: Hai! 👋 
I think max 3 would be okay. Let's go with that 👍

Comment (brentswisher) [reply packages/components/src/disabled/stories/index.js]: Updated :)

**Review by ItsJonQ — APPROVED** [review]
🚀 Tested and looks good to me! Thank you @brentswisher !

**Comment by ItsJonQ** [comment]
Will merge once Travis is 💚 !

**Pull request merged**